lib/Sema/MiscDiagnostics.cpp

```diff
@@ -357,6 +367,13 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
       }
     }
 
+    void checkMoveExpr(MoveExpr *moveExpr) {
+      if (!isa<DeclRefExpr>(moveExpr->getSubExpr())) {
+        Ctx.Diags.diagnose(moveExpr->getLoc(),
```

Comment:
@jckarter do you think I need to check for more things here? When I talked with @DougGregor he mentioned to check that this check also catches field accesses (which it does)... but I figured I would ask out of an abundance of caution.

Reply:
I'm not sure that we need an ironclad AST-level diagnostic here, since we will still need a diagnostic in SIL anyway to detect lvalues that can't be statically moved. So I think this is fine to reject obviously unsupportable cases.